lib/hyrax/transactions/steps/save.rb

```diff
@@ -33,6 +33,7 @@ def call(change_set, user: nil)
           begin
             new_collections = changed_collection_membership(change_set)
             unsaved = change_set.sync
+            unsaved.embargo = @persister.save(resource: unsaved.embargo) if unsaved.embargo.present?
             saved = @persister.save(resource: unsaved)
```

Comment:
Does saving the embargo to the resource here cascade to saving the resource itself? If it does, I think the following line might be redundant when `unsaved.embargo.present? == true`

Reply:
I don't believe it does. the embargo would be saved on this version of the resource, but unless we then save the resource itself...it wouldn't "stay". 

also, in the event that the resource does not have an embargo...we would still need the resource to be saved.

Reply:
True, but then you could just write something like this:

```suggestion
          if unsaved.embargo.present?
            unsaved.embargo = @persister.save(resource: unsaved.embargo)
          else
            @persister.save(resource: unsaved)
          end
```

But you'd only want to do that if you were positive that saving the embargo also saved the resource itself. Anyway, saving twice is better than never. I don't think it's a huge deal that should hold the PR up

Reply:
yeah, I don't believe that saving the embargo on the resource, saves the resource itself. and also...we're adding logic about leases here in the same way in #6111. so then this would become even longer.

thanks!